src/main/java/jenkins/scm/api/SCMEvent.java

```diff
@@ -253,13 +322,47 @@ public String toString() {
     }
 
     /**
-     * Annotation used to flag parts of the event data that comes from the event payload itself and consequently
-     * cannot be trusted as state from the SCM.
+     * Annotation used to flag parts of the event data that comes from the event payload itself.
      */
     @Documented
     @Inherited
     @Target({ElementType.METHOD, ElementType.FIELD, ElementType.LOCAL_VARIABLE})
     @Retention(RetentionPolicy.RUNTIME)
-    public @interface Untrusted {
+    public @interface EventData {
+        /**
+         * The trustability of the event data.
+         *
+         * @return The trustability of the event data.
+         */
+        Trustability value() default Trustability.PAYLOAD;
+    }
+
+    /**
+     * The trustability of the event payload data.
+     *
+     * @see #isPayloadTrusted()
+     * @see #payloadTrustability()
+     * @see EventData
+     */
+    public enum Trustability {
+        /**
+         * Signals that the {@link EventData} annotated method's return value or annotated field content is trusted.
+         */
+        TRUSTED(true),
+        /**
+         * Signals that the {@link EventData} annotated method's return value or annotated field content is untrusted.
+         */
+        UNTRUSTED(false),
+        /**
+         * Signals that the {@link EventData} annotated method's return value or annotated field content's is derived
+         * from the event payload. To determine the trustability of the data for a specific event,
+         * consult {@link #isPayloadTrusted()}.
```

Comment:
I am baffled here. `isPayloadTrusted` consults `Trustability.trusted`, which for `PAYLOAD` is always `false`.